NebulaAPI/Extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// #pragma once
+// #ifndef IEnumerableExtensions.cs_H_
+// #define IEnumerableExtensions.cs_H_
+// 
+// #endif
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NebulaAPI.Extensions;
+
+public static class IEnumerableExtensions
+{
+    public static void ForEach<T>(this IEnumerable<T> source, Action<T> action)
```

Comment:
I'm not a big fan of this extension.

There are many allocations happening implicitly, and the JIT cannot optimize any of the work done because Action is also an allocated closure because captures are generally used.

The `IEnumerable` is always going to allocate an enumerator box, which is usually avoided for the usual collections (array, list, etc) if foreach is done directly, and in many places in the APIs in the language libraries.

I'm not 100% sure the APIs bring enough convenience - especially if accidentally used in hot loops - is worth it.